src/main/java/com/epam/aidial/cfg/domain/service/InterceptorService.java

```diff
@@ -85,6 +91,63 @@ public Collection<Interceptor> getAllAtRevision(Integer revision) {
                 .collect(Collectors.toList());
     }
 
+    @Transactional
+    public void refreshInterceptorEndpoints() {
+        var interceptorEntities = interceptorJpaRepository.findByContainerIdIsNotNull();
+
+        for (var interceptorEntity : interceptorEntities) {
+            var interceptorContainerEntity = interceptorEntity.getInterceptorContainer();
+            String containerId = interceptorContainerEntity.getContainerId();
+
+            DeploymentInfoDto deploymentInfo = deploymentService.getById(containerId);
+            validateDeploymentInfo(deploymentInfo, containerId);
+
+            // TODO [VPA]: extract this logic
+            String containerUrl = deploymentInfo.getUrl();
+            String completionEndpoint = containerUrl
+                    + Optional.ofNullable(interceptorContainerEntity.getCompletionEndpointPath()).orElse("");
+            String configurationEndpoint = containerUrl
+                    + Optional.ofNullable(interceptorContainerEntity.getConfigurationEndpointPath()).orElse("");
+
+            interceptorEntity.setEndpoint(completionEndpoint);
+            interceptorEntity.setConfigurationEndpoint(configurationEndpoint);
+        }
+
+        interceptorJpaRepository.saveAll(interceptorEntities);
+    }
+
+    // TODO [VPA]: use validator class
+    private void validateDeploymentInfo(DeploymentInfoDto deploymentInfo, String containerId) {
+        if (deploymentInfo == null) {
+            throw new IllegalArgumentException("Container with ID '%s' not found in cache".formatted(containerId));
+        }
+
+        String deploymentUrl = deploymentInfo.getUrl();
+        if (StringUtils.isBlank(deploymentUrl)) {
+            throw new IllegalArgumentException(
+                "Container URL is not present, please check if it is deployed. Container ID: %s".formatted(containerId)
+            );
+        }
+    }
+
+    private void resolveEndpointsIfContainerSource(Interceptor interceptor) {
+        if (!(interceptor.getSource() instanceof InterceptorContainerSource containerSource)) {
+            return;
+        }
+
+        // TODO [VPA]: extract this logic
+        DeploymentInfoDto deploymentInfo = deploymentService.getById(containerSource.getContainerId());
+        String containerUrl = deploymentInfo.getUrl();
```

Comment:
deploymentInfo can be null here

Reply:
validator call before validated presence of deployment & url